code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,266 @@
+	burrowed = TRUE
+
+	for(var/mob/living/carbon/xenomorph/spiderling/kids in view(1, owner.loc))
+		RegisterSignal(kids, COMSIG_MOVABLE_MOVED, .proc/unburrow)
+		kids.alpha = 0
+		kids.mouse_opacity = 0
+		kids.density = FALSE
+		kids.throwpass = TRUE
+
+	RegisterSignal(X, COMSIG_MOVABLE_MOVED, .proc/unburrow)
+	succeed_activate()
+	add_cooldown()
+
+/datum/action/xeno_action/burrow/proc/unburrow(mob/M)
+	SIGNAL_HANDLER
+	if(!burrowed)
+		return
+	var/mob/living/carbon/xenomorph/X = owner
+	X.alpha = 255
+	X.mouse_opacity = 255
+	X.density = TRUE
+	X.throwpass = FALSE
+	burrowed = FALSE
+	UnregisterSignal(X, COMSIG_MOVABLE_MOVED)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Leash Ball
+// ***************************************
+
+/datum/action/xeno_action/activable/leash_ball
+	name = "leash_ball" // change to in character name later
+	ability_name = "leash_ball" // change to in character name later
+	mechanics_text = " Spit a huge ball of web that snares groups of marines "
+	action_icon_state = "scatter_spit" // temporary until I get my own icons
+	plasma_cost = 1
+	cooldown_timer = 1 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_LEASH_BALL
+
+/datum/action/xeno_action/activable/leash_ball/use_ability(atom/A)
+	var/turf/target = get_turf(A)
+	var/mob/living/carbon/xenomorph/X = owner
+	X.face_atom(target)
+	if(!do_after(X, 1 SECONDS, TRUE, X, BUSY_ICON_DANGER)) // currently here for balance prediction, shooting a 5x5 AoE snare is pretty insane even for T3 imo
+		return fail_activate()
+
+	var/datum/ammo/xeno/acid/web/leash_ball = GLOB.ammo_list[/datum/ammo/xeno/acid/web/leash_ball]
+	var/obj/projectile/newspit = new /obj/projectile(get_turf(X))
+
+	newspit.generate_bullet(leash_ball)
+	newspit.fire_at(target, X, null, newspit.ammo.max_range)
+	succeed_activate()
+	add_cooldown()
+
+/obj/structure/xeno/aoe_leash
+	name = "Snaring Web"
+	icon = 'icons/obj/items/projectiles.dmi' // temp ?
+	icon_state = "boiler_gas2" // temp
+	desc = "Looks very sticky"
+	destroy_sound = "alien_resin_break"
+	obj_integrity = 0
+	max_integrity = 1920
+	/// Radius for how far the leash should affect humans and how far away they may walk
+	var/leash_radius = 5
+	/// How much more integrity aoe_leash gains per caught marine, it is preferable that max_integrity is this var * 8.
+	var/integrity_increase = 240
+	/// List of beams to be removed on obj_destruction
+	var/list/beams
+	layer = ABOVE_ALL_MOB_LAYER
+	anchored = TRUE
+
+/// Humans caught get beamed and registered for proc/check_dist, aoe_leash also gains increased integrity for each caught human
+/obj/structure/xeno/aoe_leash/Initialize(mapload, atom/A)
+	. = ..()
+	var/mob/living/carbon/human/victims
+	for(victims in view(leash_radius, loc))
+		beam(victims, "beam_heavy", 'icons/obj/items/projectiles.dmi', INFINITY, INFINITY)
+		RegisterSignal(victims, COMSIG_MOVABLE_MOVED, .proc/check_dist)
+		obj_integrity = obj_integrity + integrity_increase
```

Comment:
should be done with a min call 
obj integrity = min(obj integr + increase, maxintegrity)